.github/workflows/ci.yml

```diff
@@ -34,14 +64,18 @@ jobs:
         with:
           crate: cargo-sort
 
-      - name: Run lints
-        uses: actions-rs/cargo@v1
+      - name: Get xtask
+        uses: actions/cache@v3
         with:
-          command: run
-          args: -p xtask --no-default-features ci lint
+          path: target/debug/xtask
+          key: xtask-${{ hashFiles('xtask/src/**', 'xtask/Cargo.toml') }}
```

Comment:
It's a bit unfortunate this has to be copy-pasted everywhere, but the only solution I know is creating more actions, which even in the [action-defined-as-YAML](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) case requires an entirely separate repo. It's really annoying that GitHub doesn't support the advanced YAML features that one would use for this w/ GitLab CI.

Reply:
Yeah I tried to look into it and couldn't find a more satisfactory solution...